### PR TITLE
Use field groups for inline editing of embedded admins

### PR DESCRIPTION
**Pull request by ethanresnick**
This diff is a bit hard to follow because it has the new features together with spacing and other tweaks. Apologies. 

But the basic idea is this: for the inline table editing mode, each field is currently given it's own cell in the table. For associated admins with more than a few fields, this makes the each `<tr>` very wide/awkward and, more importantly, it ignores the groups set up by the user in the associated admin class. This PR implements a more-usable approach, in which each cell/column holds a group rather than a single field, that way there's a clear hierarchy of *group heading* -> *group fields* as you go across each row.

### CODE REVIEW

**Comment by greg0ire** [comment]
> This diff is a bit hard to follow because it has the new features together with spacing and other tweaks. Apologies. 

Maybe you could make several commits? `git add --patch` is your friend.

**Comment by ethanresnick** [comment]
@greg0ire I haven't used Sonata in years, so I wouldn't even know where to start with this PR at this point. But I'll leave it open in case anyone else wants to run with it. Or, you're free to close it!
